src/Tokenizer.ts

```diff
@@ -148,24 +148,87 @@ export class _Tokenizer {
-      this.lexer.state.top = true;
-      const tokens = this.lexer.blockTokens(text);
-      this.lexer.state.top = top;
+      let lines = rtrim(cap[0], '\n').split('\n');
+      let raw = '';
+      let text = '';
+      const tokens: Token[] = [];
+
+      while (lines.length > 0) {
+        let inBlockquote = false;
+        const currentLines = [];
+
+        while (lines.length > 0) {
+          // get lines up to a continuation
+          if (/^ {0,3}>/.test(lines[0])) {
+            currentLines.push(lines.shift());
+            inBlockquote = true;
+          } else if (!inBlockquote) {
+            currentLines.push(lines.shift());
+          } else {
+            break;
+          }
+        }
+
+        const currentRaw = currentLines.join('\n');
+        const currentText = currentRaw
+          // precede setext continuation with 4 spaces so it isn't a setext
+          .replace(/\n {0,3}((?:=+|-+) *)(?=\n|$)/g, '\n    $1')
+          .replace(/^ {0,3}>[ \t]?/gm, '');
+        raw = raw ? `${raw}\n${currentRaw}` : currentRaw;
+        text = text ? `${text}\n${currentText}` : currentText;
+
+        // parse blockquote lines as top level tokens
+        // merge paragraphs if this is a continuation
+        const top = this.lexer.state.top;
+        this.lexer.state.top = true;
+        this.lexer.blockTokens(currentText, tokens, true);
+        this.lexer.state.top = top;
+
+        // if there is no continuation then we are done
+        if (lines.length === 0) {
+          break;
+        }
+
+        const lastToken = tokens[tokens.length - 1];
+
+        if (lastToken?.type === 'code') {
+          // blockquote continuation cannot be preceded by a code block
+          break;
+        } else if (lastToken?.type === 'blockquote') {
+          // include continuation in nested blockquote
+          const oldBlockquoteToken = lastToken as Tokens.Blockquote;
+          const newText = oldBlockquoteToken.raw + '\n' + lines.join('\n');
+          const newBlockquoteToken = this.blockquote(newText)!;
+          tokens[tokens.length - 1] = newBlockquoteToken;
+
+          raw = raw.substring(0, raw.length - oldBlockquoteToken.raw.length) + newBlockquoteToken.raw;
+          text = text.substring(0, text.length - oldBlockquoteToken.text.length) + newBlockquoteToken.text;
+          break;
+        } else if (lastToken?.type === 'list') {
+          // include continuation in nested list
+          const oldListToken = lastToken as Tokens.List;
+          const newText = oldListToken.raw + '\n' + lines.join('\n');
+          const newListToken = this.list(newText)!;
+          tokens[tokens.length - 1] = newListToken;
+
+          raw = raw.substring(0, raw.length - lastToken.raw.length) + newListToken.raw;
+          text = text.substring(0, text.length - oldListToken.raw.length) + newListToken.raw;
+          lines = newText.substring(tokens[tokens.length - 1].raw.length).split('\n');
+          continue;
+        }
```

Comment:
This section seems to have some code duplication. Would it be useful to combine the logic for `list` and `blockquote` here a bit?

Just playing around with an example without actually testing, I'm thinking something like this:

```
    if (lastToken?.type === 'blockquote' || lastToken?.type === 'list') {
      const oldToken = lastToken as Tokens.Blockquote | Tokens.List;
      const newText = oldToken.raw + '\n' + lines.join('\n');
      const newToken = this[lastToken.type]!(newText);
      tokens[tokens.length - 1] = newToken;

      raw = raw.substring(0, raw.length - oldToken.raw.length) + newToken.raw;
      text = text.substring(0, text.length - oldToken.text.length) + newToken.text;
      if (lastToken.type === 'list') {
        lines = newText.substring(tokens[tokens.length - 1].raw.length).split('\n');
      }
    }
 ```

Reply:
The `Tokens.Blockquote | Tokens.List` will mean we can't use `oldToken.text` since Tokens.List doesn't have a `text` property. And `text = text.substring(...` line is different between them.

I figured rather than peppering if statements for their differences it would be cleaner to duplicate some of the code. 

If there were more than two duplicates I think it would be worth it to create a method but DRY typically only helps when code is duplicated more than 3 times.

Reply:
In that case, what about reusing the same naming structure in each block for easier maintenance burden? The two are 90% identical, but this is difficult to see when `oldListToken` vs `oldBlockquoteToken` is making them appear more different than they are.

For me, it would be easier to visualize the differences between the two if the identical sections were also labeled identically.